### PR TITLE
Update compiler guide contributors 

### PR DESCRIPTION
**Pull request by thomasnyman**
Added recent contributors from GitHub and the C/C++ Compiler BP Calls.

It's entirely possible I've missed someone, so please add to this PR if you realize someone is missing.

As with #185 and #291 I will reach out to these individuals to confirm they consent to be named in the guide before merging.

### CODE REVIEW

**Comment by jwakely** [comment]
I gladly give my consent to be included here.

**Review by thesamesam — APPROVED** [review]
Happy to be included - thanks!

File: docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md [hunk 1]
```diff
@@ -1314,16 +1314,24 @@ The OpenSSF Developer BEST Practices Working group thanks Ericsson for their gen
 - Robert Byrne, Ericsson
 - Jussi Auvinen, Ericsson
 - Christopher "CRob" Robinson, Intel
+- Daniel Stenberg, wolfSSL
 - David A. Wheeler, Linux Foundation
 - David Edelsohn, IBM
+- Dominik Czarnota, Trail of Bits
+- Florian Berbar
+- Florian Weimar, Red Hat
```
[the remark this answers is not in the document]
Comment (thomasnyman) [reply docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md]: Thanks and my apologies. Fixed in 4a26ee2b32ba7761c5f47f0029f6a04693c20a9d.

**Comment by Flo4152** [comment]
I really enjoyed contribute to this guide and I consent to be included too. 

Thank you !

**Comment by david-a-wheeler** [comment]
@thomasnyman - thanks for updating the contributor list.

I'm honored to be included in such an illustrious list of contributors. My *sincere* thanks to all who have contributed!

**Comment by thomasnyman** [comment]
Mayank and Merve have reached out to me outside of Github and confirmed their consent. I will merge the contributor lines for everyone who's confirmed so far.

**Pull request merged**